packages/paste-core/utilities/flex/src/index.tsx

```diff
@@ -0,0 +1,191 @@
+export type BasisOptions = ResponsiveValue<string | number>;
+export type WrapOptions = ResponsiveValue<boolean | 'wrap' | 'nowrap'>;
+
+export interface FlexProps {
+  display?: DisplayOptions;
+  vertical?: VerticalOptions;
+  vAlignContent?: VerticalAlign;
+  hAlignContent?: HorizontalAlign;
+  grow?: GrowOptions;
+  shrink?: ShrinkOptions;
+  basis?: BasisOptions;
+  wrap?: WrapOptions;
+}
+
+const getGrow = ({grow}: FlexProps): {} => {
+  if (typeof grow === 'number' || Array.isArray(grow)) {
+    return grow;
+  }
+
+  if (grow) {
+    return 1;
+  }
+
+  return 0; // default
+};
+
+const getShrink = ({shrink, basis}: FlexProps): {} => {
+  if (typeof shrink === 'number' || Array.isArray(shrink)) {
+    return shrink;
+  }
+
+  if (shrink) {
+    return 1;
+  }
+
+  if (shrink === false) {
+    return 0;
+  }
+
+  if (basis && basis !== 'auto') {
+    return 0;
+  }
+
+  return 1; // default
+};
+
+const getSuffix = (item: BasisOptions): {} => {
+  const suffix = typeof item === 'number' || String(parseInt(item as string, 10)) === item ? 'px' : '';
+  return item + suffix;
+};
+
+const getBasis = ({basis}: FlexProps): {} => {
+  if (Array.isArray(basis)) {
+    return (basis as BasisOptions[]).map((value: BasisOptions) => {
+      return getSuffix(value);
+    });
+  }
+
+  if (basis) {
+    return getSuffix(basis);
+  }
+
+  return 'auto'; // default
+};
+
+const getVertical = ({vertical}: FlexProps): {} => {
+  if (Array.isArray(vertical)) {
+    return vertical;
+  }
+
+  if (vertical) {
+    return 'column';
+  }
+
+  return 'row'; // default
+};
+
+const getWrap = ({wrap}: FlexProps): {} => {
+  if (Array.isArray(wrap)) {
+    return wrap;
+  }
+
+  if (wrap) {
+    return 'wrap';
+  }
+
+  return 'nowrap'; // default
+};
+
+const RemapedVerticalAlignments = {
+  top: 'flex-start',
+  center: 'center',
+  bottom: 'flex-end',
+  stretch: 'stretch',
+};
+
+const vAlignToProps = ({vAlignContent}: FlexProps): {} => {
+  if (Array.isArray(vAlignContent)) {
+    return (vAlignContent as VerticalAlignOptions[]).map(value => RemapedVerticalAlignments[value]);
+  }
+
+  if (vAlignContent) {
+    return RemapedVerticalAlignments[vAlignContent as VerticalAlignOptions];
+  }
+
+  return 'flex-start'; // default
+};
+
+const RemapedHorizontalAlignments = {
+  left: 'flex-start',
+  center: 'center',
+  right: 'flex-end',
+  around: 'space-around',
+  between: 'space-between',
+};
+
+const hAlignToProps = ({hAlignContent}: FlexProps): {} => {
+  if (Array.isArray(hAlignContent)) {
+    return (hAlignContent as HorizontalAlignOptions[]).map(value => RemapedHorizontalAlignments[value]);
+  }
+
+  if (hAlignContent) {
+    return RemapedHorizontalAlignments[hAlignContent as HorizontalAlignOptions];
+  }
+
+  return 'flex-start'; // default
+};
+
+const getFlexStyles = (props: FlexProps): FlexboxProps => {
+  const {basis, vertical, grow, shrink, wrap, hAlignContent, vAlignContent} = props;
```

Comment:
We can simplify this by passing props into every function directly

Reply:
Similar. I think I have it right. Let me know.